### PR TITLE
fix: extend core module config interfaces

### PR DESCRIPTION
**Pull request by chenhuang444**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/plugin-proxy/src/index.ts [hunk 1]
```diff
@@ -39,8 +39,6 @@ export type ProxyOptions = string | Record<string, string>;
 
 declare module '@modern-js/core' {
   export interface DevConfig {
-    dev: {
-      proxy?: ProxyOptions;
-    };
```
Comment (leeight) [inline packages/cli/plugin-proxy/src/index.ts:44]: 之前是一个 BUg？

File: packages/server/server/src/type.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ declare module 'http' {
 
 declare module '@modern-js/core' {
   interface UserConfig {
-    bff: {
+    bff?: {
```
Comment (leeight) [inline packages/server/server/src/type.ts:17]: 还有一个 bff 的地方也需要改一下

File: packages/cli/plugin-proxy/src/index.ts (hunk 1 above)
Comment (chenhuang444) [reply packages/cli/plugin-proxy/src/index.ts:44]: 嗯，对的。只是之前实际上通过 as any，绕过了类型检测。所以这个类型定义没有起到作用

File: packages/server/server/src/type.ts (hunk 2 above)
Comment (chenhuang444) [reply packages/server/server/src/type.ts:17]: done

**Review by caohuilin — APPROVED** [review]

**Pull request merged**